service/controller/resource/monitoring/scrapeconfigs/resource.go

```diff
@@ -157,12 +179,13 @@ func toData(v interface{}, config Config) ([]byte, error) {
 	return scrapeConfigs, nil
 }
 
-func getTemplateData(cluster metav1.Object, config Config) (*TemplateData, error) {
+func getTemplateData(cluster metav1.Object, config Config, authenticationMechanism string) (*TemplateData, error) {
 	clusterID := key.ClusterID(cluster)
 
 	d := &TemplateData{
 		AdditionalScrapeConfigs:   config.AdditionalScrapeConfigs,
 		APIServerURL:              key.APIUrl(cluster),
+		AuthenticationMechanism:   authenticationMechanism,
```

Comment:
Why not calling `getAPIAuthenticationMechanism` here ?

Reply:
Done

Reply:
Nice